src/layoutparser/models/paddledetection/layoutmodel.py

```diff
@@ -0,0 +1,272 @@
+            image (np.ndarray): image (np.ndarray)
+        Returns:
+            inputs (dict): input of model
+        """
+        # read rgb image
+        image, im_info = decode_image(image, self.im_info)
+        # resize image by target_size and max_size
+        image, im_info = resize(image, im_info)
+        # normalize image
+        image, im_info = normalize_image(image, im_info)
+        # transpose images
+        image, im_info = permute(image, im_info)
+
+        inputs = {}
+        inputs['image'] = np.array((image, )).astype('float32')
+        inputs['im_shape'] = np.array((im_info['im_shape'], )).astype('float32')
+        inputs['scale_factor'] = np.array(
+            (im_info['scale_factor'], )).astype('float32')
+        return inputs
+
+    def gather_output(self, np_boxes, np_masks):
+        """process output"""
+        layout = Layout()
+        results = []
+        if reduce(lambda x, y: x * y, np_boxes.shape) < 6:
+            print('[WARNNING] No object detected.')
+            results = {'boxes': np.array([])}
+        else:
+            results = {}
+            results['boxes'] = np_boxes
+            if np_masks is not None:
+                results['masks'] = np_masks
+
+        np_boxes = results['boxes']
+        expect_boxes = (np_boxes[:, 1] > self.threshold) & (np_boxes[:, 0] > -1)
+        np_boxes = np_boxes[expect_boxes, :]
+
+        for np_box in np_boxes:
+            clsid, bbox, score = int(np_box[0]), np_box[2:], np_box[1]
+            x_1, y_1, x_2, y_2 = bbox
+
+            if self.label_map is not None:
+                label = self.label_map[clsid]
+
+            cur_block = TextBlock(
+                Rectangle(x_1, y_1, x_2, y_2), type=label, score=score
+            )
+            layout.append(cur_block)
+
+        return layout
+
+    def detect(self,
+                image):
+        """Detect the layout of a given image.
+        Args:
+            image (:obj:`np.ndarray` or `PIL.Image`): The input image to detect.
+        Returns:
+            :obj:`~layoutparser.Layout`: The detected layout of the input image
+        """
+        # Convert PIL Image Input
+        if isinstance(image, Image.Image):
+            if image.mode != "RGB":
+                image = image.convert("RGB")
+            image = np.array(image)
+
+        inputs = self.preprocess(image)
+
+        np_boxes, np_masks = None, None
+        input_names = self.predictor.get_input_names()
+        for i,input_name in enumerate(input_names):
+            input_tensor = self.predictor.get_input_handle(input_name)
+            input_tensor.copy_from_cpu(inputs[input_name])
+
+        self.predictor.run()
+        output_names = self.predictor.get_output_names()
+        boxes_tensor = self.predictor.get_output_handle(output_names[0])
+        np_boxes = boxes_tensor.copy_to_cpu()
+
+        layout = self.gather_output(np_boxes, np_masks)
+        return layout
+
+    def untar_files(self, model_tar, model_dir):
```

Comment:
It feels counter-intuitive to put this function inside the modeling folder, and I think it's better to include this part inside the PathManager, by rewriting this function [`_get_local_path`](https://github.com/facebookresearch/iopath/blob/13c016ab94e50e0eab9bea4d2e96ef1883169992/iopath/common/file_io.py#L767). So basically it will be something like this:

```python
class PaddleModelURLHandler(HTTPURLHandler):
    """
    Supports download and file check for dropbox links
    """

    def _get_supported_prefixes(self):
        return ["https://paddle-model-ecology.bj.bcebos.com"]

    def _isfile(self, path):
        return path in self.cache_map
    
    def _get_local_path(
        self,
        path: str,
        force: bool = False,
        cache_dir: Optional[str] = None,
        **kwargs: Any,
    ) -> str:
        """
        This implementation downloads the remote resource and caches it locally.
        The resource will only be downloaded if not previously requested.
        """
        self._check_kwargs(kwargs)
        if (
            force
            or path not in self.cache_map
            or not os.path.exists(self.cache_map[path])
        ):
            logger = logging.getLogger(__name__)
            parsed_url = urlparse(path)
            dirname = os.path.join(
                get_cache_dir(cache_dir), os.path.dirname(parsed_url.path.lstrip("/"))
            )
            filename = path.split("/")[-1]
            if len(filename) > self.MAX_FILENAME_LEN:
                filename = filename[:100] + "_" + uuid.uuid4().hex

            cached = os.path.join(dirname, filename)
            with file_lock(cached):
                if not os.path.isfile(cached):
                    logger.info("Downloading {} ...".format(path))
                    cached = download(path, dirname, filename=filename)
                
#### ----> Add from here 
                if path.endswith(".tar"):
                    untar_function() 

            logger.info("URL {} cached in {}".format(path, cached))
            self.cache_map[path] = cached
        return self.cache_map[path]
```

You might also need to check whether the untar'ed version exists beforehand to avoid duplicated downloading.